workflow-generator/src/main/java/com/wipro/ats/bdre/wgen/NodeCollection.java

```diff
@@ -473,19 +459,13 @@ public String toXML(Set<String> printedNodeNames) {
 
             }
         }
-        if (joinNode != null) {
-
-            if (!printedNodeNames.contains(joinNode.getName())) {
+        if (joinNode != null && !printedNodeNames.contains(joinNode.getName())) {
                 stringBuilder.append(joinNode);
                 printedNodeNames.add(joinNode.getName());
-            }
         }
-        if (haltStepNode != null) {
-            if (!printedNodeNames.contains(haltStepNode.getName())) {
+        if (haltStepNode != null && !printedNodeNames.contains(haltStepNode.getName())) {
                 stringBuilder.append(haltStepNode);
                 printedNodeNames.add(haltStepNode.getName());
-            }
-
         }
         if (termStepNode != null) {
             if (!printedNodeNames.contains(termStepNode.getName())) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Merge this if statement with the enclosing one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.openbdre.org/coding_rules#rule_key=squid%3AS1066)